### PR TITLE
Clean meta plugin tests from CPU/GPU plugin

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Move BATCH related test out from CPU/GPU func test to BATCH func test
 - Move HETERO related test out from CPU/GPU func test to HETERO func test
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_plugin/caching_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+#include "ov_ops/multiclass_nms_ie_internal.hpp"
+#include "ov_ops/nms_ie_internal.hpp"
+#include "ov_ops/nms_static_shape_ie.hpp"
+
+using namespace ov;
+using namespace ov::test::behavior;
+
+namespace {
+static const std::vector<ov::element::Type> precisionsCPU = {
```
Comment (praasz) [inline src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_plugin/caching_tests.cpp:15]: Should be `CPU` used in names in shared tests?
Can be some generic name?

File: src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_infer_request/gpu_dyn_batch_shape_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        ov::test::utils::removeFilesWithExt(cacheFolderName, "blob");
+        ov::test::utils::removeFilesWithExt(cacheFolderName, "cl_cache");
+        ov::test::utils::removeDir(cacheFolderName);
+    }
+}
+
+auto hetero_config = []() {
+    return ov::AnyMap{{ov::device::priorities.name(), ov::test::utils::DEVICE_GPU}};
+};
```
Comment (praasz) [inline src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_infer_request/gpu_dyn_batch_shape_tests.cpp:116]:
```suggestion
const auto hetere_config = testing::Values(ov::AnyMap{{ov::device::priorities.name(), ov::test::utils::DEVICE_GPU}});
```
Can use similar approach in other test parameters, instead make vectors copy to values.

File: src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_plugin/caching_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+//
+
+#include "behavior/ov_plugin/caching_tests.hpp"
+
+#include "ov_ops/multiclass_nms_ie_internal.hpp"
+#include "ov_ops/nms_ie_internal.hpp"
+#include "ov_ops/nms_static_shape_ie.hpp"
+
+using namespace ov;
+using namespace ov::test::behavior;
```
Comment (praasz) [inline src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_plugin/caching_tests.cpp:12]:
```suggestion
namespace ov {
namespace test {
namespace behavior {
```
?

File: src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp [hunk 4]
```diff
@@ -30,4 +31,49 @@ INSTANTIATE_TEST_SUITE_P(smoke_Hetero_BehaviorTests,
                          ::testing::Combine(::testing::Values(ov::test::utils::DEVICE_HETERO),
                                             ::testing::ValuesIn(hetero_properties)),
                          OVPropertiesTests::getTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(smoke_HeteroOVGetMetricPropsTest, OVGetMetricPropsTest, ::testing::Values("HETERO"));
```
Comment (yangwang201911) [inline src/plugins/hetero/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp:35]:
```suggestion
INSTANTIATE_TEST_SUITE_P(smoke_HeteroOVGetMetricPropsTest, OVGetMetricPropsTest, ::testing::Values(ov::test::utils::DEVICE_HETERO));
```

**Review by yangwang201911 — APPROVED** [review]

**Pull request merged**